### ISSUE TITLE
Cop idea: detect redundant setting to empty string inside string interpolation

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#13121]
## Is your feature request related to a problem? Please describe.

I sometimes encounter code with an if/else condition that sets a variable to an empty string as one of the outcomes. 

This isn't needed if the variable is only being used in string interpolation, and although thats not always something rubocop can detect, I assume it can detect if it the condition is inside of a string.

```ruby
# bad, but rubocop isn't able to know that.
example = [true, false].sample ? '--example-arg' : ''
puts "program #{example} --bonus"

# manually, this can be corrected if the user can ensure this is the only usage of example.
puts "program #{'--example-arg' if [true, false].sample} --bonus"
```

```ruby
# bad
puts "program #{example = [true, false].sample ? '--example-arg' : ''} --bonus"

# good
puts "program #{'--example-arg' if [true, false].sample} --bonus"
```

## Describe the solution you'd like

For the following code patterns to be flagged as offenses and autocorrectable:
```ruby
# bad
"bar #{condition ? 'foo' : ''}"
"bar #{if condition; 'foo' else '' end}"

# good
"bar #{'foo' if condition}"
```

(and also the same for cases where the ternary operator is reversed and it should correct to `unless` instead of `if`, and also for empty strings in the form of `""`-- not sure about whether flagging assigning things to `nil` is safe, i think it is?)

**Comment by Zopolis4** [comment]
commenting to remove the stale label (i would still like this feature, and my pr to implement it is fully functional)

**Comment by Zopolis4** [comment]
not stale, see above
